src/geo/projection/projection.js

```diff
@@ -98,6 +98,14 @@ export default class Projection {
         }
     }
 
+    isPointAboveHorizon(tr: Transform, p: Point): boolean {
```

Comment:
:+1: that's a nice way to fork the behavior and do that only when needed.

I think it's also important that we reject similar queries for points that fall outside the globe on `map.getElevation` queries.

Reply:
> I think it's also important that we reject similar queries for points that fall outside the globe on map.getElevation queries.

@karimnaaji Can you elaborate on this and why it is the desired behavior?

Reply:
For similar reasons as it is a desired behavior for feature querying, refer https://github.com/mapbox/mapbox-gl-native-internal/issues/3343 for `map.getElevation`. These querying APIs should adopt _what you see is what you get_ (WYSIWYG), otherwise we fail at a 1:1 matching between what you're actually querying and what you're getting back. Why would we ever want screen points outside the globe to resolve elevation incorrectly at points on the horizon?

Reply:
Good idea @karimnaaji, that makes sense! Adding a helper should be as simple as publicly exposing  `isPointAboveHorizon`, though the latter approach would save a raytracing call so might improve performance. Let's keep this in mind for a future feature!